pkg/content/discover.go

```diff
@@ -52,6 +86,36 @@ func AllPageTitles(pages map[string]*Page) []string {
 	return allPageTitles
 }
 
+func PagesByCategory(pages map[string]*Page) map[string][]*Page {
+	pagesByCategory := make(map[string][]*Page)
+
+	for _, page := range pages {
+		for _, category := range page.Meta.Categories {
+			pagesByCategory[category] = append(pagesByCategory[category], page)
+		}
+	}
+
+	return pagesByCategory
+}
+
+func AllCategories(pages map[string]*Page) []string {
+	categories := map[string]struct{}{}
+	for _, page := range pages {
+		for _, category := range page.Meta.Categories {
+			categories[category] = struct{}{}
+		}
+	}
+
+	keys := make([]string, len(categories))
+	i := 0
+	for k := range categories {
+		keys[i] = k
+		i++
+	}
```

Comment:
Rather than manually managing `i`, can this just append to an empty slice?

```suggestion
	keys := make([]string)
	for k := range categories {
		keys = append(keys, k)
	}
```